Neos.Flow/Classes/Http/Helper/ResponseInformationHelper.php

```diff
@@ -149,7 +150,18 @@ public static function prepareHeaders(ResponseInterface $response): array
     public static function makeStandardsCompliant(ResponseInterface $response, RequestInterface $request): ResponseInterface
     {
         $statusCode = $response->getStatusCode();
-        if ($request->hasHeader('If-Modified-Since') && $response->hasHeader('Last-Modified') && $statusCode === 200) {
+        if ($request->hasHeader('If-None-Match') && in_array($request->getMethod(), ['HEAD', 'GET'])
+            && $response->hasHeader('ETag') && $statusCode === 200) {
```

Comment:
Should we also handle `If-None-Match` with another request method to return a `412` status? (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match)

Reply:
Not sure we can do that in the standard compliance component as this is 
 used to prevent conflicting write operations see: https://tools.ietf.org/html/rfc7232#section-3.2. 

That can imho only be solved by the application itself.

Reply:
Well, right, that would need to happen in the cache handling component to check the match condition and the request handling short-circuited at that place. So, something for FullPageCache ;)

Reply:
Full Page Cache only needs to ensure that the ETag header is sent after this change went in. As  the 304 status will be sent from here.

Nice thing is that every Flow Application that creates proper ETags benefits. Regardless wether the same result was calculated again or cached.

While recalculating and later removing the body may look like a pointless approach it makes sense from the client perspective as the client can reuse the already parsed result from the cache and can safe transfer and parsing time.

Reply:
I added a separate pr to remove the 412 status rendering https://github.com/neos/flow-development-collection/pull/1979  

Will probably take some time and maybe even have to wait for flow 7